### PR TITLE
Reorganize to comply with Arduino IDE 1.5 Library Specification

### PR DESCRIPTION
**Pull request by adamvoss**
Reorganize to comply with Arduino IDE 1.5 Library Specification

Specification can be found at https://github.com/arduino/Arduino/wiki/Arduino-IDE-1.5:-Library-specification.

Note: I just arbitrarily started the version at 1.0.0, since I did not see any existing version identifier.

### CODE REVIEW

**Comment by iwanders** [comment]
Thanks for putting up this PR.

> Note: I just arbitrarily started the version at 1.0.0, since I did not see any existing version identifier.

Yeah, I don't really bother with version numbers for my own libraries, I usually embed the git hash into my firmware at compile time ([like this](https://github.com/iwanders/focus_stacking/blob/master/firmware/git_version.sh)). I usually have the libraries as a git submodule, which are at a specific commit for the parent repository's commit. So that way I can always discern which version is running if.

Using version `1.0.0` or higher means that the public API should be considered [stable](http://semver.org/#spec-item-4). Which I guess we can just say it is, I have used a project that relied on this library almost daily for over a year and didn't run into issues / bugs. Since I don't have access to an OBD9141 port right now I certainly won't be changing the public API. Judging from the [faq](http://semver.org/#how-do-i-know-when-to-release-100) entry we are definitely past `1.0.0`!

File: library.properties [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name=OBD9141
+version=1.0.0
+author=Ivor Wanders
+maintainer=Ivor Wanders
+sentence=Read data from the the ISO 9141-2 (K-line) pin of an OBD-II port found in cars.
+paragraph=This makes it easy to read data from the diagnostic port and display or record it.
```
Comment (iwanders) [inline library.properties]: The paragraph line is apparently [prefixed](https://github.com/arduino/Arduino/wiki/Arduino-IDE-1.5:-Library-specification#libraryproperties-file-format) by the sentence, also the library itself does not display or record data, it surely helps if that is the intent of the user though, I get what you are trying to say.

Perhaps something like this describes it well:
`paragraph=This provides a class that makes it easier to interact with the port, handling the initialization and provides methods to retrieve PID values from the ECU.`

File: library.properties [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+category=Communication
+url=https://github.com/iwanders/OBD9141
+architectures=*
+includes=OBD9141.h
```
Comment (iwanders) [inline library.properties]: Please add a newline after the last line.

**Comment by adamvoss** [comment]
All your suggestions sounded good.  I did a force-push that should addressed them.

**Comment by iwanders** [comment]
Thanks for your contribution.

**Comment by iwanders** [comment]
Was the zip file intentional? I didn't spot that in the first version of the PR... It also contains a `.git` folder, but the repository in the zip file itself also has unstaged changes.

I presume this was included by accident (and I didn't spot that on the second review pass). Am I good to delete that [zipfile](https://github.com/iwanders/OBD9141/blob/master/OBD9141.zip)?

**Comment by adamvoss** [comment]
Darn, I'm really sorry about that!  Yeah, I created that for testing the library.properties file and it it accidentally was added when I did the fixes.

As far as I'm concerned, you could even go ahead and re-write to get rid of it even if you are concerned about repo size and such.

**Comment by iwanders** [comment]
No problem, mistakes happen. Next time its better to add additional commits to the PR, and then they can be squashed (if need be) just before the merge.

>As far as I'm concerned, you could even go ahead and re-write to get rid of it even if you are concerned about repo size and such.

Even though that's not the prefered way to rewrite the history on the master branch once it has gone public I've done just that. Given that the other version of the branch was out there for just 5 minutes I think it's highly unlikely that someone already pulled the changes and has now ended up with a diverged repository. So I deleted the zip file, ammended the merge commit and force pushed the changes. Thanks for the quick response to resolve this.

**Pull request merged**